src/util/GeneratorConverters.h

```diff
@@ -0,0 +1,39 @@
+#include <optional>
+
+#include "util/Generator.h"
+#include "util/Iterators.h"
+
+template <typename ValueType>
+cppcoro::generator<ValueType> fromInputRange(
+    ad_utility::InputRangeTypeErased<ValueType> inputRange) {
+  for (auto& value : inputRange) {
+    co_yield value;
+  }
+}
+
+template <typename ValueType>
+ad_utility::InputRangeTypeErased<ValueType> fromGenerator(
+    cppcoro::generator<ValueType> generator) {
+  struct Generator : ad_utility::InputRangeFromGet<ValueType> {
+    Generator(cppcoro::generator<ValueType> generator)
+        : generator{std::move(generator)},
+          iterator{this->generator.begin()},
+          end{this->generator.end()} {}
+
+    std::optional<ValueType> get() override {
+      if (iterator != end) {
+        auto value{std::move(*iterator)};
+        iterator++;
+        return value;
+      }
```

Comment:
This seems to be completely unnecessary, as the deduction guide of `InputRangeTypeErased` should handle this, s.t. you can write
`auto x = InputRangeTypeErased(std::move(someGenerator))`  and the value_type will be deduced automatically.